src/os/filestore/FileStore.cc

```diff
@@ -3795,20 +3799,24 @@ int FileStore::_move_tempobjs(const coll_t& src_cid, const ghobject_t& src_oid,
       }
   }
 
-  dout(10) << "move_tempobjs "  << cid << "/" << oid << " "  <<  " = " << r << dendl;
+  dout(10) << "merge_delete_srcobj "  << cid << "/" << oid << " "  <<  " = " << r << dendl;
 
   out3:
+    if (dstcmp > 0) {      // if dstcmp == 0 the guard already says "in-progress"
+      _set_replay_guard(**b, spos, &oid);
```

Comment:
Does this usage of _set_replay_guard allow _check_replay_guard to return 0?  Does this design ensure that the source object is actually removed given that you set the replay guard before removing the object?